packages/domains-table/src/domains-table/domains-table-registered-until-cell.tsx

```diff
@@ -0,0 +1,18 @@
+import { PartialDomainData } from '@automattic/data-stores';
+import { useLocale } from '@automattic/i18n-utils';
+
+interface DomainsTableRegisteredUntilCellProps {
+	domain: PartialDomainData;
+}
+
+export const DomainsTableRegisteredUntilCell = ( {
+	domain,
+}: DomainsTableRegisteredUntilCellProps ) => {
+	const localeSlug = useLocale();
+
+	return domain.has_registration
+		? new Intl.DateTimeFormat( localeSlug, { dateStyle: 'medium' } ).format(
```

Comment:
Nice, all the browsers we care about support `format()`. Better than adding a `moment` dependency to our package!